lib/spack/spack/package_base.py

```diff
@@ -1897,7 +1897,8 @@ def _test_deprecated_warning(self):
     # TODO (post-34236): run_test, etc.
     def test(self):
         # Defer tests to virtual and concrete packages
-        warnings.warn(self._test_deprecated_warning)
+        # Ensure passing warn() a string for python 3.6/rhel8,ubuntu,etc.
+        warnings.warn(str(self._test_deprecated_warning))
```

Comment:
(Bug) Is this a typo? I don't understand why we want to cast to `str`. Isn't the failure due to the fact that either:
1. `_test_deprecated_warning` was not declared as a property
2. `_test_deprecated_warning` is not getting called

?

Reply:
🤦